ipaserver/secrets/client.py

```diff
@@ -73,10 +77,14 @@ def __init__(
     def init_creds(self):
         name = gssapi.Name(self.client_service,
                            gssapi.NameType.hostbased_service)
-        store = {'client_keytab': self.keytab,
-                 'ccache': 'MEMORY:Custodia_%s' % b64encode(
-                     os.urandom(8)).decode('ascii')}
-        return gssapi.Credentials(name=name, store=store, usage='initiate')
+        ccname = "MEMORY:Custodia_{}".format(
+            b64encode(os.urandom(8)).decode('ascii'))
+        store = {'client_keytab': self.keytab, 'ccache': ccname}
+        creds = gssapi.Credentials(name=name, store=store, usage='initiate')
+        if self.set_krb5ccname:
+            # do this _after_ 'creds' was successfully constructed
+            os.environ['KRB5CCNAME'] = ccname
```

Comment:
I would have preferred if you could refactor this use of CustodiaClient.init_creds as a context manager like I did with ipatests/test_cmdline/test_ipagetkeytab.py. So far, I can see that most of CustodiaClient instance consumers simply call fetch_key() or just call a constructor. I would prefer do switch that to a code that does something like

```
client = CustodiaClient(..)
with client.init_creds() as cli:
    cli.fetch_key()
```
This would make the context under which execution happens clearly defined and will reduce potential time we change `KRBCCNAME`.